.gitignore

```diff
@@ -1,11 +1,622 @@
-/*/nbproject/private/
-/*/build/
-/*/dist/
-/*/lib/
-/*/test/
-
-# Idea
-*.iml
-*.ipr
+#################
```

Comment:
This ignore is an overkill ;)

Reply:
I can remove something if you want

Reply:
Make it java specific + the IDEs

Reply:
I've removed CMake things and Visual Studio, if you think I should remove anything else just let me know